src/Raven.Server/Documents/Handlers/AI/Agents/ConversationDocument.cs

```diff
@@ -197,6 +242,9 @@ public static List<BlittableJsonReaderObject> GenerateTools(JsonOperationContext
         List<BlittableJsonReaderObject> tools = [];
         foreach (var q in configuration.Queries ?? [])
         {
+            if (q.Options is { AllowModelQueries: false })
```

Comment:
The logic goes:
* If you specified nothing, we add you.
* If you specified `AddToInitialQuery = true`, then you need to explicitly state `AllowModelQueries = true`

Reply:
done